test/gtest/typecheck_test.cpp

```diff
@@ -0,0 +1,62 @@
+/*
```

Comment:
This is more or less what the compiler is doing now.
It would be much easier to add a flag for the compiler to stop compiling after a certain pass.